### ISSUE TITLE
Weird compilation error behaviour after test generation

### ISSUE THREAD

**Issue by MaelAudren** [STAMP-project/botsing#35]
#### Characteristics
- **Issue Type**: [bug]
- **Reproducibility**: [always]
- **Severity**: [minor]
- **Tool/Service/Component**: [botsing-reproduction,1.0.1]
- **Execution Environment**: [linux]
- **Reporter**: [Mael Audren, mael.audren@activeeon.com]

#### Description
When executing Botsing the error `Failed to compile test case` is displayed on the output. It seems that the error is written on the output without making crash the original thread indeed the process end without error.
Moreover it does not prevent to generate the test.

#### Steps to reproduce

Follow the README.md of the project https://github.com/STAMP-project/botsing-gradle-plugin
When generating the example the issue will appear.

#### Error log
```
Diagnostic: package org.junit does not exist: 8
Diagnostic: package org.junit does not exist: 9
Diagnostic: package org.evosuite.runtime does not exist: 10
Diagnostic: package org.evosuite.runtime does not exist: 12
Diagnostic: package org.evosuite.runtime does not exist: 13
Diagnostic: package org.junit.runner does not exist: 14
Diagnostic: package org.evosuite.runtime.annotation does not exist: 9
Diagnostic: package org.junit does not exist: 10
Diagnostic: package org.junit does not exist: 11
Diagnostic: package org.junit does not exist: 12
Diagnostic: package org.junit does not exist: 13
Diagnostic: package org.evosuite.runtime.sandbox does not exist: 14
Diagnostic: package org.evosuite.runtime.sandbox.Sandbox does not exist: 15
Diagnostic: cannot find symbol
  symbol: class EvoSuiteClassExclude: 17
Diagnostic: cannot find symbol
  symbol: class RunWith: 16
Diagnostic: cannot find symbol
  symbol: class EvoRunnerParameters: 16
Diagnostic: package org.evosuite.runtime.vnet does not exist: 21
Diagnostic: package org.evosuite.runtime.thread does not exist: 25
Diagnostic: cannot find symbol
  symbol:   class Test
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest: 19
Diagnostic: package org.junit does not exist: 20
Diagnostic: cannot find symbol
  symbol:   class BeforeClass
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest_scaffolding: 28
Diagnostic: cannot find symbol
  symbol:   class AfterClass
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest_scaffolding: 43
Diagnostic: cannot find symbol
  symbol:   class Before
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest_scaffolding: 49
Diagnostic: cannot find symbol
  symbol:   class After
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest_scaffolding: 61
Diagnostic: package org.evosuite.runtime.vnet does not exist: 21
Diagnostic: package org.evosuite.runtime.thread does not exist: 25
Diagnostic: package org.evosuite.runtime.thread does not exist: 25
Diagnostic: package org.evosuite.runtime does not exist: 30
Diagnostic: package org.evosuite.runtime does not exist: 31
Diagnostic: package org.evosuite.runtime does not exist: 32
Diagnostic: package org.evosuite.runtime does not exist: 33
Diagnostic: package org.evosuite.runtime does not exist: 34
Diagnostic: package org.evosuite.runtime does not exist: 35
Diagnostic: package org.evosuite.runtime.sandbox.Sandbox does not exist: 35
Diagnostic: package org.evosuite.runtime.sandbox does not exist: 36
Diagnostic: package org.evosuite.runtime.classhandling does not exist: 37
Diagnostic: package org.evosuite.runtime does not exist: 40
Diagnostic: cannot find symbol
  symbol:   variable Sandbox
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest_scaffolding: 45
Diagnostic: package org.evosuite.runtime.jvm does not exist: 53
Diagnostic: package org.evosuite.runtime.sandbox does not exist: 54
Diagnostic: package org.evosuite.runtime does not exist: 56
Diagnostic: package org.evosuite.runtime does not exist: 57
Diagnostic: package org.evosuite.runtime.agent does not exist: 58
Diagnostic: package org.evosuite.runtime.jvm does not exist: 64
Diagnostic: package org.evosuite.runtime.classhandling does not exist: 65
Diagnostic: package org.evosuite.runtime.sandbox does not exist: 67
Diagnostic: package org.evosuite.runtime.agent does not exist: 68
Diagnostic: package org.evosuite.runtime does not exist: 69
Diagnostic: package org.evosuite.runtime.classhandling does not exist: 86
Diagnostic: cannot find symbol
  symbol:   method fail(java.lang.String)
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest: 25
Diagnostic: cannot find symbol
  symbol:   method verifyException(java.lang.String,java.lang.ArithmeticException)
  location: class eu.stamp.botsing.Fraction_0_tmp__ESTest: 31
RegularFileObject[/tmp/EvoSuite_0_1543404432703/eu/stamp/botsing/Fraction_0_tmp__ESTest.java]
1: /*
2:  * This file was automatically generated by EvoSuite
3:  * Wed Nov 28 12:27:12 CET 2018
4:  */
5: 
6: package eu.stamp.botsing;
7: 
8: import org.junit.Test;
9: import static org.junit.Assert.*;
10: import static org.evosuite.runtime.EvoAssertions.*;
11: import eu.stamp.botsing.Fraction;
12: import org.evosuite.runtime.EvoRunner;
13: import org.evosuite.runtime.EvoRunnerParameters;
14: import org.junit.runner.RunWith;
15: 
16: @RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, useJEE = true) 
17: public class Fraction_0_tmp__ESTest extends Fraction_0_tmp__ESTest_scaffolding {
18: 
19:   @Test(timeout = 4000)
20:   public void test0()  throws Throwable  {
21:       Fraction fraction0 = new Fraction(1694, (-1));
22:       // Undeclared exception!
23:       try { 
24:         fraction0.getShiftedValue(1);
25:         fail("Expecting exception: ArithmeticException");
26:       
27:       } catch(ArithmeticException e) {
28:          //
29:          // / by zero
30:          //
31:          verifyException("eu.stamp.botsing.Fraction", e);
32:       }
33:   }
34: }
RegularFileObject[/tmp/EvoSuite_0_1543404432703/eu/stamp/botsing/Fraction_0_tmp__ESTest.java]
1: /**
2:  * Scaffolding file used to store all the setups needed to run 
3:  * tests automatically generated by EvoSuite
4:  * Wed Nov 28 12:27:12 CET 2018
5:  */
6: 
7: package eu.stamp.botsing;
8: 
9: import org.evosuite.runtime.annotation.EvoSuiteClassExclude;
10: import org.junit.BeforeClass;
11: import org.junit.Before;
12: import org.junit.After;
13: import org.junit.AfterClass;
14: import org.evosuite.runtime.sandbox.Sandbox;
15: import org.evosuite.runtime.sandbox.Sandbox.SandboxMode;
16: 
17: @EvoSuiteClassExclude
18: public class Fraction_0_tmp__ESTest_scaffolding {
19: 
20:   @org.junit.Rule 
21:   public org.evosuite.runtime.vnet.NonFunctionalRequirementRule nfr = new org.evosuite.runtime.vnet.NonFunctionalRequirementRule();
22: 
23:   private static final java.util.Properties defaultProperties = (java.util.Properties) java.lang.System.getProperties().clone(); 
24: 
25:   private org.evosuite.runtime.thread.ThreadStopper threadStopper =  new org.evosuite.runtime.thread.ThreadStopper (org.evosuite.runtime.thread.KillSwitchHandler.getInstance(), 3000);
26: 
27: 
28:   @BeforeClass 
29:   public static void initEvoSuiteFramework() { 
30:     org.evosuite.runtime.RuntimeSettings.className = "eu.stamp.botsing.Fraction"; 
31:     org.evosuite.runtime.GuiSupport.initialize(); 
32:     org.evosuite.runtime.RuntimeSettings.maxNumberOfThreads = 100; 
33:     org.evosuite.runtime.RuntimeSettings.maxNumberOfIterationsPerLoop = 10000; 
34:     org.evosuite.runtime.RuntimeSettings.mockSystemIn = true; 
35:     org.evosuite.runtime.RuntimeSettings.sandboxMode = org.evosuite.runtime.sandbox.Sandbox.SandboxMode.RECOMMENDED; 
36:     org.evosuite.runtime.sandbox.Sandbox.initializeSecurityManagerForSUT(); 
37:     org.evosuite.runtime.classhandling.JDKClassResetter.init();
38:     setSystemProperties();
39:     initializeClasses();
40:     org.evosuite.runtime.Runtime.getInstance().resetRuntime(); 
41:   } 
42: 
43:   @AfterClass 
44:   public static void clearEvoSuiteFramework(){ 
45:     Sandbox.resetDefaultSecurityManager(); 
46:     java.lang.System.setProperties((java.util.Properties) defaultProperties.clone()); 
47:   } 
48: 
49:   @Before 
50:   public void initTestCase(){ 
51:     threadStopper.storeCurrentThreads();
52:     threadStopper.startRecordingTime();
53:     org.evosuite.runtime.jvm.ShutdownHookHandler.getInstance().initHandler(); 
54:     org.evosuite.runtime.sandbox.Sandbox.goingToExecuteSUTCode(); 
55:     setSystemProperties(); 
56:     org.evosuite.runtime.GuiSupport.setHeadless(); 
57:     org.evosuite.runtime.Runtime.getInstance().resetRuntime(); 
58:     org.evosuite.runtime.agent.InstrumentingAgent.activate(); 
59:   } 
60: 
61:   @After 
62:   public void doneWithTestCase(){ 
63:     threadStopper.killAndJoinClientThreads();
64:     org.evosuite.runtime.jvm.ShutdownHookHandler.getInstance().safeExecuteAddedHooks(); 
65:     org.evosuite.runtime.classhandling.JDKClassResetter.reset(); 
66:     resetClasses(); 
67:     org.evosuite.runtime.sandbox.Sandbox.doneWithExecutingSUTCode(); 
68:     org.evosuite.runtime.agent.InstrumentingAgent.deactivate(); 
69:     org.evosuite.runtime.GuiSupport.restoreHeadlessMode(); 
70:   } 
71: 
72:   public static void setSystemProperties() {
73:  
74:     java.lang.System.setProperties((java.util.Properties) defaultProperties.clone()); 
75:     java.lang.System.setProperty("user.dir", "/home/mael/stamp/wp3/botsing-plugin/example"); 
76:     java.lang.System.setProperty("file.encoding", "UTF-8"); 
77:     java.lang.System.setProperty("java.io.tmpdir", "/tmp"); 
78:     java.lang.System.setProperty("user.country", "FR"); 
79:     java.lang.System.setProperty("user.home", "/home/mael"); 
80:     java.lang.System.setProperty("user.language", "fr"); 
81:     java.lang.System.setProperty("user.name", "mael"); 
82:     java.lang.System.setProperty("user.timezone", ""); 
83:   }
84: 
85:   private static void initializeClasses() {
86:     org.evosuite.runtime.classhandling.ClassStateSupport.initializeClasses(Fraction_0_tmp__ESTest_scaffolding.class.getClassLoader() ,
87:       "eu.stamp.botsing.Fraction"
88:     );
89:   } 
90: 
91:   private static void resetClasses() {
92:   }
93: }

Failed to compile test case:
Fraction fraction0 = new Fraction(1694, (-1));
fraction0.getShiftedValue(1);
```

**Comment by pderakhshanfar** [comment]
This error happens when Botsing cannot save the final generated test. This issue should be fixed in the latest release.

**Comment by xdevroey** [comment]
Fixed in #31 I close this issue too.
